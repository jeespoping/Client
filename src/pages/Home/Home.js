import React, { useState } from "react";
import Client from "../../components/Client/Client";
import ClientForm from "../../components/ClientForm/ClientForm";

import "./Home.scss";

export default function Home() {
  const [selectedComponent, setSelectedComponent] = useState("form");
  const [client, setClient] = useState(null);

  const handleComponent = () => {
    switch (selectedComponent) {
      case "form":
        return (
          <ClientForm
            setClient={setClient}
            setSelectedComponent={setSelectedComponent}
          />
        );
      case "client":
        return (
          <Client client={client} setSelectedComponent={setSelectedComponent} />
        );

      default:
        return (
          <ClientForm
            setClient={setClient}
            setSelectedComponent={setSelectedComponent}
          />
        );
    }
  };

  return <div className="home">{handleComponent()}</div>;
}
