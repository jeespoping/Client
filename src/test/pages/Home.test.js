import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import Home from "../../pages/Home";

describe("Prueba en <Home />", () => {
  test("Debe mostrar correctamente", () => {
    let wrapper = shallow(<Home />);

    expect(wrapper).toMatchSnapshot();
  });
});
