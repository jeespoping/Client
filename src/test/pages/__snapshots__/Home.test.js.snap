// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <Home /> Debe mostrar correctamente 1`] = `
<div
  className="home"
>
  <ClientForm
    setClient={[Function]}
    setSelectedComponent={[Function]}
  />
</div>
`;
